src/main/java/EventsAndCommands/UtilityCommands/QuoteCommand.java

```diff
@@ -6,53 +6,60 @@
         this.category = Categories.Utility;
     }
 
     @Override
     protected void execute(CommandEvent commandEvent) {
         String args = commandEvent.getArgs();
-        String messageId = args;
-        boolean isFromDifferentChannel = false;
+        String messageId = args.substring(0, 18);
+        String caption = args.length() >= 19 ? args.substring(19) : "";
         final boolean hasAMentionedChannel = !commandEvent.getMessage().getMentionedChannels().isEmpty();
 
         if (hasAMentionedChannel) {
             quoteChannel = commandEvent.getMessage().getMentionedChannels().get(0);
             mention = commandEvent.getMessage().getMentionedChannels().get(0).getAsMention();
-            messageId = args.replaceAll(mention, "").strip();
-
         } else {
             quoteChannel = commandEvent.getChannel();
         }
 
         try {
 
-
             quoteChannel.getMessageById(messageId).queue(x -> {
 
                 final OffsetDateTime creationTime = x.getCreationTime();
                 final Member member = x.getMember();
-
                 EmbedBuilder embed = new EmbedBuilder();
 
+                if(x.getContentRaw().isEmpty()){ // when the user quotes a quote or something else that makes no sense like those messages with no messages whatsoever
+                    embed.setDescription("you can't do that!");
+                    embed.setColor(Color.RED);
+                    commandEvent.reply(embed.build(), success -> commandEvent.getMessage().delete().queue());
+                    return;
+                }
 
                 embed.setDescription(x.getContentRaw() + "\n \n [[Jump to message]](" + x.getJumpUrl() + ")  ");
-                embed.appendDescription("Creation time:  " + creationTime.toLocalDate().format(DateTimeFormatter.ISO_LOCAL_DATE) + "\n[Quoted by " + commandEvent.getMember().getAsMention() + "]");
-                embed.setColor(Color.ORANGE);
-
+                if (caption.isEmpty()) {
+                    embed.appendDescription("Creation time:  " + creationTime.toLocalDate().format(DateTimeFormatter.ISO_LOCAL_DATE) + "\n[Quoted by " + commandEvent.getMember().getAsMention() + "]");
+                } else if (caption.length() > MessageEmbed.TEXT_MAX_LENGTH) { // discord won't allow more than 2000, but it's better to make sure just in case.
+                    embed.appendDescription("Creation time:  " + creationTime.toLocalDate().format(DateTimeFormatter.ISO_LOCAL_DATE) + "\n[Quoted by " + commandEvent.getMember().getAsMention() + "]\n your caption is too long (longer than 2048 characters chill for a sec)");
+                } else {
+                    embed.appendDescription("Creation time:  " + creationTime.toLocalDate().format(DateTimeFormatter.ISO_LOCAL_DATE)
+                            + "\nQuoted by " + commandEvent.getMember().getAsMention() + " with the caption:\n" + caption);
+                }
 
                 try {
```

Comment:
nested try/catch? I'd suggest you use the `isEmpty()` functionality to check if it has anything.

Reply:
~F4nt0M

Reply:
if there's no image, `x.getAttachments().get(0)` will throw an exception, i tried to fix it before pushing but i couldn't because i'm not familiar with JDA.

Reply:
I'll have to check it out at home. I think I fixed it in #5 using `if(!yadda.yadda().isEmpty()){` and then using get `get(0)` command. I did not remove the try catch tho - that is merely due to the fact that I left it in in case I missed something.